plugins/com.google.cloud.tools.eclipse.usagetracker.test/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManagerWithServerTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.usagetracker;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.mockito.Matchers.anyBoolean;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.when;
+
+import com.google.cloud.tools.eclipse.preferences.AnalyticsPreferences;
+import com.google.cloud.tools.eclipse.test.util.http.TestHttpServer;
+import com.google.cloud.tools.eclipse.usagetracker.AnalyticsPingManager.PingEvent;
+import com.google.cloud.tools.eclipse.util.CloudToolsInfo;
+import java.util.Map;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences;
+import org.eclipse.swt.widgets.Display;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AnalyticsPingManagerWithServerTest {
+
+  @Mock private IEclipsePreferences preferences;
+  @Mock private Display display;
+  @Mock private ConcurrentLinkedQueue<PingEvent> pingEventQueue;
```

Comment:
Currently there is no test to see if the job is picking up events. Tests in this class are to see if events are being queued and/or the opt-in dialog is presented depending on the opt-in states. The mock queue also prevents actually making HTTP requests in this test class.